exercises/binary-search-tree/canonical-data.json

```diff
@@ -0,0 +1,177 @@
+{
+    "exercise": "binary-search-tree",
+    "version": "1.0.0",
+    "comments": [
+        "Each test case assumes an empty/new tree.",
+        "As per exercism/problem-specifications#996 key 'tree_data' counts as an input",
+        "to test generators.",
+        "The key 'tree_data' represents an array of numbers for which the data should be ",
+        "inserted/added to the tree as it appears in the array from left to right.",
+        "e.g. tree_data: ['2', '1', '3', '6', '7', '5']",
+        "Insert 2. Insert 1. Insert 3. Insert 6, so on...",
+        "This canonical-data does not restrict the data type of array elements to either",
+        "strings or integers.",
+        "The key 'expected' represents tree state as JSON object of schema :",
+        "{",
+        "    'title':'nodeObject',",
+        "    'type':'object',",
+        "    'properties':{",
+        "        'data':{",
+        "            'type':'string'",
+        "        },",
+        "        'left':{",
+        "            'type':'nodeObject'",
+        "        },",
+        "        'right':{",
+        "            'type':'nodeObject'",
+        "        }",
+        "    },",
+        "    'required':['data', 'left', 'right']",
+        "}"
+    ],
+    "cases": [{
+            "description": "data is retained",
+            "property": "data",
+            "input": {
+                "tree_data": ["4"]
```

Comment:
@coriolinus @Insti 
I just saw issue #987 
Should the input key `tree_data` be changed to lowerCamelCase

Reply:
From what I understand #996 is not allowing additional properties. so does #987 make sense?

Reply:
I have a mild preference for `snake_case` over `lowerCamelCase`, but TBH I didn't even think about the applicability of #987 here. 

As I understand it, the driving intent behind that issue was to avoid using keys which require special handling because they contain spaces or otherwise are not permissible as variable names. As that's handled by either naming convention, I don't have a preference one way or the other about whether you should update your cases here.